artemis-server/src/main/resources/schema/artemis-configuration.xsd

```diff
@@ -2290,6 +2290,14 @@
             </xsd:documentation>
          </xsd:annotation>
       </xsd:attribute>
+      <xsd:attribute name="no-message-forwarding" type="xsd:boolean" use="optional" default="false">
+         <xsd:annotation>
+            <xsd:documentation>
+               If this is true, the mirror at the opposite end of the link will not forward messages coming from that link to any other mirrors down the line.
+               This is false by default.
+            </xsd:documentation>
+         </xsd:annotation>
+      </xsd:attribute>
```

Comment:
This name+description, plus my earlier note about the acknowledgements, make me realise it is still going to re-mirror other things like address and queue creations, and more importantly deletions. Which could still be quite problematic just like the message-duping themselves, likely worse due to timing interactions of creation and deletion.

I'm not sure anything can be done about that except only using pre-created everything. I am again sceptical of this topology.

Reply:
Got that fixed.